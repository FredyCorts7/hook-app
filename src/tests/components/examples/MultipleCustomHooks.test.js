import React from 'react';
import { shallow } from 'enzyme';
import MultipleCustomHooks from '../../../components/examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
import '@testing-library/jest-dom';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Testing to <MultipleCustomHooks />', () => {
  useCounter.mockReturnValue({
    counter: 0,
    increment: () => {},
  });

  test('should be the same component in initial state', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show information', () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: 'Camila',
          quote: 'Everybody ignore to me',
        },
      ],
      loading: false,
      error: null,
    });

    const wrapper = shallow(<MultipleCustomHooks />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.alert').exists()).toBeFalsy();
    expect(wrapper.find('.mb-0').text().trim()).toBe('Everybody ignore to me');
    expect(wrapper.find('footer').text().trim()).toBe('Camila');
  });
});
