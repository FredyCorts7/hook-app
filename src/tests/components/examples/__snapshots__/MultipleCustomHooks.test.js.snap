// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <MultipleCustomHooks /> should be the same component in initial state 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alrt alert-info text-center"
  >
    Loading...
  </div>
</div>
`;

exports[`Testing to <MultipleCustomHooks /> should show information 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-0"
    >
      Everybody ignore to me
    </p>
    <footer
      className="blockquote-footer"
    >
      Camila
    </footer>
  </blockquote>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <button
        className="btn btn-warning col-6"
        onClick={[Function]}
      >
        Previous quote
      </button>
      <button
        className="btn btn-primary col-6"
        onClick={[Function]}
      >
        Next quote
      </button>
    </div>
  </div>
</div>
`;
